src/processing/ParameterizedOperationsUtil.java

```diff
@@ -3,195 +3,435 @@
+	 */
+	public static BoardObject colorChange(BoardObject obj, UserId id, Intensity intensity) {
+
+		BoardObject newObj = null;
+		try {
+
+			newObj = colorChangeUtil(obj, id, intensity);
+			stackUtil(newObj);
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: operation completed"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: operation failed!"
+			);
+		}
+
+		try {
+
+			// To send all the pixel updates to UI
+			CommunicateChange.provideChanges(newObj.getPrevIntensity(), newObj.getPixels());
+
+			// To send selection updates to UI
+			CommunicateChange.identifierToHandler.get(CommunicateChange.identifierUI).giveSelectedPixels(newObj.getPixels());
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: updates sent to UI"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: UI updates failed!"
+			);
+		}
+
+	// change `boardOp` of old object and return it
+	IBoardObjectOperation newBoardOp = new ColorChangeOperation(intensity);
+	obj.setOperation(newBoardOp);
+	obj.setUserId(newObj.getUserId());
+
+	return obj;
+	}
+
+	/**
+	 * Function for implementing rotation of select-able objects.
+	 */
+	public static BoardObject rotation(BoardObject obj, UserId id, Angle angleOfRotation) {
+
+		BoardObject newObj = null;
+		try {
+
+			newObj = rotationUtil(obj, id, angleOfRotation);
+			stackUtil(newObj);
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotation: operation completed"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotation: operation failed!"
+			);
+		}
+
+		try {
+
+			// To send all the pixel updates to UI
+			CommunicateChange.provideChanges(newObj.getPrevIntensity(), newObj.getPixels());
+
+			// To send selection updates to UI
+			CommunicateChange.identifierToHandler.get(CommunicateChange.identifierUI).giveSelectedPixels(newObj.getPixels());
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotation: updates sent to UI"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotation: UI updates failed!"
+			);
+		}
+
+	// change `boardOp`, `UserId` of old object and return it
+	IBoardObjectOperation newBoardOp = new RotateOperation(angleOfRotation);
```

Comment:
431, 432, 433, 435 lines should be indented to the level of other statements in the function (4 spaces in this case)